ethereum/ethevents/events.go

```diff
@@ -290,13 +290,36 @@ func (ev *EthereumEvents) SubscribeEthereumEventLogs(ctx context.Context, fromBl
 		go ev.runEventProcessor(ctx)
 	}
 
+	// check eth connection
+	connCh := make(chan bool, 1)
+	go func(chan bool) {
+		for {
+			time.Sleep(time.Second * 5)
+			blockTctx, cancel := context.WithTimeout(ctx, types.EthereumReadTimeout*2)
+			if _, err := ev.VotingHandle.EthereumClient.BlockByNumber(blockTctx, nil); err != nil {
+				log.Warnf("cannot check ethereum connection while running the event processor, %s", err)
+				connCh <- false
+				cancel()
+				return
+			}
+		}
+	}(connCh)
+
 	for {
 		select {
-		case err := <-sub.Err():
+		case <-sub.Err():
 			// TODO: @jordipainan do not fatal here, handle error on event subscription channel
-			log.Fatalf("ethereum events subscription error on channel: %s", err)
+			log.Warn("ethereum events subscription error on channel")
+			log.Infof("restarting ethereum events subscription with web3: %+v", ev.DialAddrs)
+			ev.EventProcessor.eventProcessorRunning = false
+			return
 		case event := <-logs:
 			ev.EventProcessor.Events <- event
+		case <-connCh:
```

Comment:
we have two identical cases in this select now; why do we have two channels instead of one?

Reply:
discussed privately and solved

Reply:
btw, the fallthrough clause cannot be applied here afaik. The compiler complains about it and I only solved it with a weird if clause.

Reply:
You're right, I forgot that fallthroughs can't be used in selects. I guess not worth doing anything more magic here, but we could at least put both cases next to each other.